### PR TITLE
feat: add redis state callbacks

### PR DESCRIPTION
**Pull request by stakach**
where core is expected to updated redis on behalf of the driver

Adds a new callback `on_redis` that passes back the following params

```crystal
is_status : Bool,  # Should we be setting the key_name in the module status hash
module_id : String,
key_name : String,
status_value : String? # delete key from redis if the value is `nil`, otherwise set the key
```

This where callbacks are currently configured in `core`
https://github.com/PlaceOS/core/blob/fa974045e82e3af1b04e94ed6fcc7caaa7dfa600/src/placeos-core/module_manager.cr#L321

The difference between status and not status: (Status is a redis hash)
* Status: https://github.com/PlaceOS/driver/blob/cd927949489823a06e55d23e1b3494ff8190b088/src/placeos-driver/storage.cr#L37
* Not Status: https://github.com/PlaceOS/driver/blob/cd927949489823a06e55d23e1b3494ff8190b088/src/placeos-driver.cr#L33

### CODE REVIEW

File: src/placeos-driver/protocol/management.cr [hunk 1]
```diff
@@ -448,6 +451,16 @@ class PlaceOS::Driver::Protocol::Management
       mod_id = request.id
       setting_name, setting_value = Tuple(String, YAML::Any).from_yaml(request.payload.not_nil!)
       on_setting.call(mod_id, setting_name, setting_value)
+    when "status"
```
Comment (caspiano) [inline src/placeos-driver/protocol/management.cr:454]: maybe we should use a string backed enum for this?

**Review by caspiano — APPROVED** [review]
looks good to me!

File: src/placeos-driver/protocol/management.cr (hunk 1 above)
Comment (stakach) [reply src/placeos-driver/protocol/management.cr:454]: Yeah will update this

File: spec/protocol_management_spec.cr [hunk 2]
```diff
@@ -30,6 +30,12 @@ describe PlaceOS::Driver::Protocol::Management do
     }))
     manager.running?.should eq(true)
 
+    # TODO:: implement this
```
Comment (caspiano) [inline spec/protocol_management_spec.cr]:
```suggestion
```

**Review by caspiano — APPROVED** [review]
looks neat!

**Comment by stakach** [comment]
this is safe to merge currently as is not a breaking change

**Pull request merged**